NetworthManager.js

```diff
@@ -0,0 +1,207 @@
+const { calculateNetworth, calculateItemNetworth } = require('./helper/calculateNetworth');
+const { parseItems, postParseItems } = require('./helper/parseItems');
+const { NetworthError, PricesError, ItemsError } = require('./helper/errors');
+const NetworthTypes = require('./NetworthTypes');
+const axios = require('axios');
+const { setItems } = require('./constants/itemsMap');
+
+class NetworthManager {
+    /**
+     * Create a new NetworthManager instance
+     * @param {Object} options - Options for the NetworthManager
+     * @param {NetworthTypes} [options.networthType=NetworthTypes.Normal] - The type of networth calculation to use
+     * @param {boolean} [options.cachePrices=true] - Whether to cache the prices for 5 minutes after fetching them or fetch them every time
+     * @param {number} [options.pricesRetries=3] - The amount of retries to fetch the prices when failing to fetch them
+     * @param {number} [options.itemsRetries=3] - The amount of retries to fetch the items when failing to fetch them
+     * @param {number} [options.itemsInterval=1000 * 60 * 60 * 12] - The interval to fetch the items from the Hypixel API
+     * @param {boolean} [options.onlyNetworth=false] - Whether to only return the total networth or the items as well
+     * @param {boolean} [options.stackItems=true] - Whether to stack items with the same name and price
+     * @param {boolean} [options.includeItemData=false] - Whether to include the item data as a property in the item object
+     */
+    constructor({ networthType, cachePrices, pricesRetries, itemsRetries, itemsInterval, onlyNetworth, stackItems, includeItemData }) {
```

Comment:
Maybe the options could be grouped into categories? Kind of like this:

```js
constructor({ networthType, onlyNetworth, stackItems, includeItemData}, { cachePrices, pricesRetries }, { itemsRetries, itemsInterval }) {
```

Also defaults?

```js
constructor({ networthType: NetworthTypes.Normal, onlyNetworth: false, stackItems: true, includeItemData: false } = { networthType: NetworthTypes.Normal, onlyNetworth: false, stackItems: true, includeItemData: false }, { cachePrices: true, pricesRetries: 3 } = { cachePrices: true, pricesRetries: 3 }, { itemsRetries: 3, itemsInterval: 1000 * 60 * 60 * 12 } = { itemsRetries: 3, itemsInterval: 1000 * 60 * 60 * 12 }) {
```

Also should probably add `pricesInterval`

Reply:
Defaults should already be added if you look at the lines under that

Reply:
@MattTheCuber just got to this and the reason I did not add this is because the prices are obtained upon getPrices call. I can add how long the prices should be cached for tho if you want me to